### PR TITLE
Revision of Reinforcement Module

### LINKED ISSUE #145: Revision of The Reinforcement Module I
- [x] Fix: General instabilities... (e.g. #66)
- [x] Change: Search should no longer depend on the module tree, but solely on configs.
- [x] Change: Introduce a cache.
- [x] ~Feature: Customized filling of vehicles (e.g. #17)~ (moved to #213)

### PR DESCRIPTION
**Pull request by oOKexOo**
**When merged this pull request will:**
- Fix #145. 
- Add Achilles_fnc_returnChildren and replace BIS_fnc_returnChildren with it.
- Extend Achilles_fnc_ShowChooseDialog.

### CODE REVIEW

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_ShowChooseDialog.sqf [hunk 1]
```diff
@@ -106,10 +109,15 @@ private _titleVariableIdentifier = format ["Ares_ChooseDialog_DefaultValues_%1",
 			// Create the combo box
 			private _ctrl_cb = _dialog ctrlCreate ["RscCombo", BASE_IDC_CTRL + _forEachIndex, _ctrl_group];
 			_ctrl_cb ctrlSetPosition [COMBO_COLUMN_X, _yCoord+LABEL_COMBO_DELTA_Y, COMBO_WIDTH, COMBO_HEIGHT];
+			private _use_data = false;
 			{
 				_x params [["_entry_text_L","",[""]], ["_entry_text_R","",[""]], ["_STR_AMAE_data","",[""]]];
 				private _id = _ctrl_cb lbAdd _entry_text_L;
 				_ctrl_cb lbSetTextRight [_id, _entry_text_R + " "];
+				if (not (_STR_AMAE_data isEqualTo "")) then
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_ShowChooseDialog.sqf]: you should replace _STR_amae_data to just _strData. It's due to my faulty regex prefixes.

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_returnChildren.sqf [hunk 2]
```diff
@@ -0,0 +1,22 @@
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// AUTHOR: 			Kex
+// DATE: 			12/26/17
+// VERSION: 		AMAE.0.1.0
+// DESCRIPTION:		Similar to BIS_fnc_returnChildren, but more efficient and dept is always 0.
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_returnChildren.sqf]: depth

File: @AresModAchillesExpansion/addons/modules_f_ares/Reinforcements/functions/fn_ReinforcementsCreateUnits.sqf [hunk 3]
```diff
@@ -2,22 +2,155 @@
 
 #define FIRST_SPECIFIC_LZ_OR_RP_OPTION_INDEX 4
 
-//FIRST_SPECIFIC_LZ_OR_RP_OPTION_INDEX = 4;
 disableSerialization;
 
-private _spawnPosition = position _logic;
+private _spawn_position = position _logic;
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/modules_f_ares/Reinforcements/functions/fn_ReinforcementsCreateUnits.sqf]: you had to didn't you? :D

File: @AresModAchillesExpansion/addons/modules_f_ares/Reinforcements/functions/fn_ReinforcementsCreateUnits.sqf [hunk 4]
```diff
@@ -2,22 +2,155 @@
...
 
-private _side_names = ["OPFOR","BLUEFOR",localize "STR_AMAE_INDEPENDENT"];
-private _sides = [east,west,independent];
+	// get all vehicles
+	{
+		private _type = configName _x;
+		private _side_id = [_x, "side", 4] call BIS_fnc_returnConfigEntry;
+		if ([_x, "scope", 0] call BIS_fnc_returnConfigEntry == 2 and _side_id <= 2) then
+		{
+			if (_type isKindOf "Tank" or _type isKindOf "Car" or _type isKindOf "Helicopter" or _type isKindOf "Plane"or _type isKindOf "Ship") then
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/modules_f_ares/Reinforcements/functions/fn_ReinforcementsCreateUnits.sqf]: space between "Plane" and or

File: @AresModAchillesExpansion/addons/modules_f_ares/Reinforcements/functions/fn_ReinforcementsCreateUnits.sqf [hunk 5]
```diff
@@ -29,47 +162,44 @@ private _dialogResult =
...
-
-//Get dialog results
-private _side = _sides select (_dialogResult select 0);
-private _infantryGroup = Ares_var_reinforcement_infantry_group;
-private _vehicleType = Ares_var_reinforcement_vehicle_class;
-private _dialogVehicleBehaviour = _dialogResult select 4;
-private _dialogLzAlgorithm = _dialogResult select 5;
-private _dialogRpAlgorithm = _dialogResult select 8;
-private _dialogUnitBehaviour = _dialogResult select 9;
+_dialogResult params ["_side_id","_veh_fac_id","_veh_cat_id","_veh_id","_veh_beh","_lzdz_algorithm","_lzdz_type","_grp_fac_id","_grp_id","_rp_algorithm","_grp_beh"];
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/modules_f_ares/Reinforcements/functions/fn_ReinforcementsCreateUnits.sqf]: The variables are short, yes, but fac, beh, cat makes no sense at a quick glance. I'd prefer more descriptive variables.

File: @AresModAchillesExpansion/addons/modules_f_ares/Reinforcements/functions/fn_ReinforcementsCreateUnits.sqf [hunk 6]
```diff
@@ -203,41 +333,43 @@ if (count _allRps > 0) then
...
 {
 	["ACE3 or AR is not loaded!"] call Achilles_fnc_showZeusErrorMessage;
 	{deleteVehicle _x} forEach _infantry_list;
 };
 
+// print a confirmation
 if (count _allRps > 0) then
 {
-	[objNull, "Transport dispatched to LZ. Squad will head to RP."] call bis_fnc_showCuratorFeedbackMessage;
+	[objNull, localize "STR_AMAE_REINFORCEMENT_DISPATCHED"] call Ares_fnc_showZeusMessage;
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/modules_f_ares/Reinforcements/functions/fn_ReinforcementsCreateUnits.sqf]: objNull isn't an parameter for it

File: @AresModAchillesExpansion/addons/modules_f_ares/Reinforcements/functions/fn_ReinforcementsCreateUnits.sqf (hunk 3 above)
Comment (oOKexOo) [reply @AresModAchillesExpansion/addons/modules_f_ares/Reinforcements/functions/fn_ReinforcementsCreateUnits.sqf]: lol

File: @AresModAchillesExpansion/addons/ui_f/functions/dialogs/fn_RscDisplayAttributes_CreateReinforcement.sqf [hunk 7]
```diff
@@ -93,10 +92,9 @@ switch (_mode) do
 		private _ctrl_veh = _dialog displayCtrl IDC_COMBO_VEHICLE;
 		lbClear _ctrl_veh;
 		{
-			private _lb_id = _ctrl_veh lbAdd getText ( _x >> "displayName");
-			private _type = configName _x;
-			_ctrl_veh lbSetData [_lb_id, _type];
-			_ctrl_veh lbSetTextRight [_lb_id, "(" + str (([_type, true] call BIS_fnc_crewCount) - ([_type, false] call BIS_fnc_crewCount)) + ") "];
+			private _lb_id = _ctrl_veh lbAdd getText (configFile >> "CfgVehicles" >> _x >> "displayName");
+			_ctrl_veh lbSetData [_lb_id, _x];
+			_ctrl_veh lbSetTextRight [_lb_id, "(" + str (([_x, true] call BIS_fnc_crewCount) - ([_x, false] call BIS_fnc_crewCount)) + ") "];
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/ui_f/functions/dialogs/fn_RscDisplayAttributes_CreateReinforcement.sqf]: joinString

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_ShowChooseDialog.sqf [hunk 8]
```diff
@@ -111,14 +111,14 @@ private _titleVariableIdentifier = format ["Ares_ChooseDialog_DefaultValues_%1",
 			_ctrl_cb ctrlSetPosition [COMBO_COLUMN_X, _yCoord+LABEL_COMBO_DELTA_Y, COMBO_WIDTH, COMBO_HEIGHT];
 			private _use_data = false;
 			{
-				_x params [["_entry_text_L","",[""]], ["_entry_text_R","",[""]], ["_STR_AMAE_data","",[""]]];
+				_x params [["_entry_text_L","",[""]], ["_entry_text_R","",[""]], ["_str_data","",[""]]];
 				private _id = _ctrl_cb lbAdd _entry_text_L;
 				_ctrl_cb lbSetTextRight [_id, _entry_text_R + " "];
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_ShowChooseDialog.sqf]: joinString?

**Pull request merged**